src/site/stages/build/plugins/add-asset-hashes.js

```diff
@@ -1,69 +1,85 @@
-function addAssetHashes(isPreview = false) {
-  // In non-development modes, we add hashes to the names of asset files in order to support
-  // cache busting. That is done via WebPack, but WebPack doesn't know anything about our HTML
-  // files, so we have to replace the references to those files in HTML and CSS files after the
-  // rest of the build has completed. This is done by reading in a manifest file created by
-  // WebPack that maps the original file names to their hashed versions. Metalsmith actions
-  // are passed a list of files that are included in the build. Those files are not yet written
-  // to disk, but the contents are held in memory.
+/* eslint-disable no-continue, no-param-reassign */
 
-  return (files, metalsmith, done) => {
-    // Read in the data from the manifest file.
-    const manifestKey = Object.keys(files).find(
-      filename => filename.match(/file-manifest.json$/) !== null,
-    );
+const environments = require('../../../constants/environments');
 
-    const originalManifest = JSON.parse(files[manifestKey].contents.toString());
+const FILE_MANIFEST_FILENAME = 'generated/file-manifest.json';
+
+function copyAssetsToTeamSitePaths(buildOptions, files, entryNamesDictionary) {
+  // TeamSite pages such as "benefits.va.gov" have hardcoded references to certain
+  // JavaScript bundles so that we can inject our header/footer into pages outside
+  // of our source. Therefore, we can't apply a hash to those bundles.
+  const teamsiteInjectionBundles = {
+    'proxy-rewrite.js': 'generated/proxy-rewrite.entry.js',
+    'vendor.js': 'generated/vendor.entry.js',
+    'polyfills.js': 'generated/polyfills.entry.js',
+    'styleConsolidated.css': 'generated/styleConsolidated.css',
+    'static-pages.css': 'generated/static-pages.css',
+  };
 
-    // The manifest contains the original filenames without the addition of .entry
-    // on the JS files. This finds all of those and modifies them to add .entry.
-    const manifest = {};
-    Object.keys(originalManifest).forEach(originalManifestKey => {
-      const matchData = originalManifestKey.match(/(.*)\.js$/);
-      if (matchData !== null) {
-        const newKey = `${matchData[1]}.entry.js`;
-        manifest[newKey] = originalManifest[originalManifestKey];
+  for (const fileManifestKey of Object.keys(teamsiteInjectionBundles)) {
+    const teamSitePath = teamsiteInjectionBundles[fileManifestKey];
+    const hashedFileName = entryNamesDictionary.get(fileManifestKey);
+
+    if (!hashedFileName) {
+      if (buildOptions.entry) {
+        // A command arg was passed to build only certain areas of the site, so
+        // it makes that it would be missing.
```

Comment:
This would mean we have a production or staging build with an `--entry` flag.
:thinking: That's weird, but probably alright.

Reply:
No, I think `--entry` is only used for localhost builds.  I don't know why this would happen. I'm just going by the comment that was there on `addAssetHashes`.